### PR TITLE
Deprecate FluentUIHostingController in iOS 16

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

There were some new APIs in iOS 16.4 that we could use instead of our neat swizzling, but it turns out we just don't need the whole class anymore.

### Binary change

Total increase: 2,480 bytes
Total decrease: -2,160 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 30,487,960 bytes | 30,488,280 bytes | ⚠️ 320 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| ControlHostingView.o | 61,712 bytes | 63,600 bytes | ⚠️ 1,888 bytes |
| __.SYMDEF | 4,604,016 bytes | 4,604,512 bytes | ⚠️ 496 bytes |
| PeoplePicker.o | 291,784 bytes | 291,880 bytes | ⚠️ 96 bytes |
| SearchBar.o | 418,824 bytes | 418,776 bytes | 🎉 -48 bytes |
| ActivityIndicatorCell.o | 80,424 bytes | 80,368 bytes | 🎉 -56 bytes |
| MSFCardNudge.o | 41,016 bytes | 40,912 bytes | 🎉 -104 bytes |
| MSFNotification.o | 143,600 bytes | 143,488 bytes | 🎉 -112 bytes |
| MSFAvatar.o | 29,176 bytes | 29,064 bytes | 🎉 -112 bytes |
| MSFPersonaButtonCarousel.o | 35,720 bytes | 35,608 bytes | 🎉 -112 bytes |
| MSFActivityIndicator.o | 31,800 bytes | 31,688 bytes | 🎉 -112 bytes |
| MSFAvatarGroup.o | 29,680 bytes | 29,560 bytes | 🎉 -120 bytes |
| MSFHeadsUpDisplay.o | 35,056 bytes | 34,928 bytes | 🎉 -128 bytes |
| MSFPersonaButton.o | 29,504 bytes | 29,344 bytes | 🎉 -160 bytes |
| MSFIndeterminateProgressBar.o | 29,880 bytes | 29,712 bytes | 🎉 -168 bytes |
| FocusRingView.o | 783,656 bytes | 782,728 bytes | 🎉 -928 bytes |
</details>

### Verification

Tested on iOS 14.5, 15.0, 16.0, and 16.2
Made sure not to regress the fixes added in 
 - #745 
 - #759 

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| FluentUIHosting_AvatarGroup_Before | FluentUIHosting_AvatarGroup_After |
| FluentUIHosting_NavBar_Before | FluentUIHosting_NavBar_After |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1690)

### CODE REVIEW

**Comment by huwilkes** [comment]
Looks like the nav bar in 15 regressed, investigating fix alternative fix

**Comment by huwilkes** [comment]
> Looks like the nav bar in 15 regressed, investigating fix alternative fix

Went with deprecation, and using just `UIHostingController` in iOS 16. Once we drop iOS 14 and 15, we should be able to just delete the file again.

File: ios/FluentUI/Core/ControlHostingView.swift [hunk 1]
```diff
@@ -99,7 +95,15 @@ open class ControlHostingView: UIView {
         )
     }
 
-    private let hostingController: FluentUIHostingController = .init(rootView: AnyView(EmptyView()))
+    private let hostingController: UIHostingController = {
+        if #unavailable(iOS 16) {
```
Comment (mischreiber) [inline ios/FluentUI/Core/ControlHostingView.swift:99]: Why 16, if we aren't API gated?

Comment (huwilkes) [reply ios/FluentUI/Core/ControlHostingView.swift:99]: The API we wanted to replace our swizzling with is only available in 16.2, but we don't need the entire class in 16. We do still need it in iOS 14 and 15, though. So I assume there are some changes to `UIHostingController` only in 16 that fix what we have work arounds for.

Comment (mischreiber) [reply ios/FluentUI/Core/ControlHostingView.swift:99]: Ohh that's wild, I didn't realize that. Can you lightly document that as a comment inline?

Comment (harrieshin) [reply ios/FluentUI/Core/ControlHostingView.swift:102]: wait, if it is 16.2 shouldn't it be 16.2 explicitly? not #unavailable (ios 16)?

Comment (huwilkes) [reply ios/FluentUI/Core/ControlHostingView.swift:102]: The new API we wanted was in 16.2, but the fix for the bugs we were working around is in just 16. Tested on a 16.0 simulator.

Comment (huwilkes) [reply ios/FluentUI/Core/ControlHostingView.swift:102]: Also, I was saying the wrong version for when the APIs were added. They were added in 16.4.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**